### PR TITLE
Add evaluation status bar item

### PR DESCRIPTION
**Pull request by stefan-toubia**
## What has Changed?

- Add a new status bar item for evaluation status

I've been bitten more than a few times because I missed the fact that I encountered an evaluation error, either because the bottom of my output channel was not visible or the output panel was collapsed.  This adds a simple status indicator to make it idiot-proof.

#### No evaluation
image

#### Evaluating (animated)
image image image

#### Success
image

#### Error
image

#### Action
Clicking the icon will open the output panel.  I couldn't get it to do more than this, I didn't put too much effort into it, it would be nice to make it toggle and ensure that the Calva Says output is shown.

## My Calva PR Checklist

I have:

- [ ] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [ ] Made sure I am directing this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [ ] Made sure I am changed the default PR base branch, so that it is not `master`. (Sorry for the nagging.)
- [ ] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.) You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. (For now you'll need to opt in to the CircleCI _New Experience_ UI to see the Artifacts tab, because bug.)
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
- [ ] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [ ] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [ ] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [ ] Created the issue I am fixing/addressing, if it was not present.
- [ ] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.

## The Calva Team PR Checklist:

Before merging we (at least one of us) have:

- [ ] Made sure the PR is directed at the `dev` branch (unless reasons).
- [ ] Read the source changes.
- [ ] Given feedback and guidance on source changes, if needed. (Please consider noting extra nice stuff as well.)
- [ ] Tested the VSIX built from the PR (well, if this is a PR that changes the source code.)
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
- [ ] If need be, had a chat within the team about particular changes.

Ping @pez, @kstehn, @cfehse

### CODE REVIEW

File: src/evaluate.ts [hunk 1]
```diff
@@ -191,18 +197,21 @@ async function loadFile(document, callback: () => { }, pprintOptions: PrettyPrin
...
             } else {
+                setEvalStatus(EvaluationStatus.none);
                 chan.appendLine("No results from file evaluation.");
             }
         }).catch((e) => { 
             chan.appendLine(`Evaluation of file ${fileName} failed: ${e}`);
+            setEvalStatus(EvaluationStatus.error);
         });
     }
     if (callback) {
```
Comment (stefan-toubia) [inline src/evaluate.ts]: This was recently added, I'm not familiar with the purpose.  Would it make sense for this to also be represented in the status bar indicator?

Comment (PEZ) [reply src/evaluate.ts]: I think this might be from @cfehse 's fix for the load file callback. And yes, would probably be material for a status indicator.

Comment (cfehse) [reply src/evaluate.ts]: Yes it is.

Comment (cfehse) [reply src/evaluate.ts]: The purpose was solely to show that an error occurred instead of ignoring the error condition completely.

**Comment by PEZ** [comment]
I think this PR addresses a big and important issue with Calva. We should really do this. However, I don't think adding more status bar items is the way to do it. How about we instead:

* use the `clj` indicator
* Make it red when there are errors
  * Put the error message in the hover.
  * Add an action to it that shows the stack trace in the REPL window. 
* animate its color during evaluation, say pulsate between grey and white (if possible)

This indicator sometimes is a button already. For toggling `cljc` repl between `clj` and `cljs`. So for the stack trace action we will need to show a menu offering both actions. I think we should then change it so that it always shows a menu. We can move the actions for opening the REPL windows there.

**Comment by stefan-toubia** [comment]
@PEZ fair enough about the status bar items, there are enough of them already.  I like the plan of using the `clj` indicator.  This should be a quick change over, I will do some refactoring to the status bar update function to clean this up a bit.

> Add an action to it that shows the stack trace in the REPL window.
I think what would make sense here is to implement this as a separate PR after #471 is finished, which implements some changes to outputting messages and stacktrace to the REPL window, which should make this very straightforward to implement.

**Comment by PEZ** [comment]
> > Add an action to it that shows the stack trace in the REPL window.
> 
> I think what would make sense here is to implement this as a separate PR after #471 is finished, which implements some changes to outputting messages and stacktrace to the REPL window, which should make this very straightforward to implement.

Makes sense. It was that PR that got me to think about it. Let's just have the error message in the tooltip then for now?

**Comment by stefan-toubia** [comment]
I'm closing this PR since it is stale, will re-open a new one when I have made more progress.

**Pull request closed without merging**